### PR TITLE
suggestion for change of deserializer (1)

### PR DESCRIPTION
**Pull request by seagetch**
This patch suggest new deserialization which uses two phase finalizer.
finalizer is decomposed into two step. restruct and finalize.
restruct change the position of the object in the tree. finalize maintains the link.
By introducing this steps, we can dynamically change the structure of the model after loading.

This helps reconstruction of the model structure, and saves into new format.

### CODE REVIEW

File: source/inochi2d/core/automation/package.d [hunk 1]
```diff
@@ -95,6 +95,9 @@ struct AutomationBinding {
         return null;
     }
 
+    void reconstruct(Puppet puppet) {
```
Comment (LunaTheFoxgirl) [inline source/inochi2d/core/automation/package.d]: for empty functions I'd prefer to have the format be a one-liner like so
```d
void reconstruct(Puppet puppet) { }
```

File: source/inochi2d/core/animation/animation.d [hunk 2]
```diff
@@ -241,6 +246,9 @@ public:
         return 0;
     }
 
+    void reconstruct(Puppet puppet) {
```
Comment (LunaTheFoxgirl) [inline source/inochi2d/core/animation/animation.d]: ditto

File: source/inochi2d/core/param/binding.d [hunk 3]
```diff
@@ -322,12 +327,19 @@ public:
         return null;
     }
 
+    override
+    void reconstruct(Puppet puppet) {
```
Comment (LunaTheFoxgirl) [inline source/inochi2d/core/param/binding.d]: ditto, for overrides do still keep the override on a seperate line
```d
override
void reconstruct(Puppet puppet) { }
```

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]
Minor stylistic requirements

**Comment by seagetch** [comment]
Adjusted style.

**Comment by LunaTheFoxgirl** [comment]
LGTM

**Pull request merged**